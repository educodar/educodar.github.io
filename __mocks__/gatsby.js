const React = require("react")
const gatsby = jest.requireActual("gatsby")

const gatsbyMock = {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest
    .fn()
    .mockImplementation(
      ({
        activeClassName,
        activeStyle,
        getProps,
        innerRef,
        partiallyActive,
        ref,
        replace,
        to,
        ...rest
      }) =>
        React.createElement("a", {
          ...rest,
          href: to,
        })
    ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn(),
}

gatsbyMock.StaticQuery.mockImplementation(({ render }) =>
  render({
    site: {
      siteMetadata: {
        title: "Educodar - Aulas de programação grátis para jovens",
        titleTemplate: "%s · Educodar",
        siteUrl: "https://educodar.com.br",
        description:
          "Projeto de ensino de programação básica e exposição a T.I. para estudantes de escolas de comunidades. Iniciativa sem fins lucrativos. Cursos e alimentação gratuitos.",
        image: "/images/logo.png",
      },
    },
  })
)

module.exports = gatsbyMock
