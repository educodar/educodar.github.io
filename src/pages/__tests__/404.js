import React from "react"
import renderer from "react-test-renderer"
import { StaticQuery } from "gatsby"

import NotFoundPage from "../404"

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({ site: { siteMetadata: {} } })
  )
})

describe("404 page", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<NotFoundPage />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
