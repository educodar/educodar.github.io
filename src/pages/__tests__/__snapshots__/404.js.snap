// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 page renders correctly 1`] = `
<article
  className="MuiBox-root MuiBox-root-2"
>
  <h1
    className="MuiTypography-root MuiTypography-h1"
  >
    Página não encontrada
  </h1>
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
  >
    Ops! A página que você está procurando pode ter sido removida ou renomeada.
  </p>
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
  >
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href="/"
      onBlur={[Function]}
      onFocus={[Function]}
      target="_self"
    >
      Voltar à página inicial
    </a>
  </p>
</article>
`;
